lighthouse-core/lib/asset-saver.js

```diff
@@ -102,8 +101,15 @@ async function saveArtifacts(artifacts, basePath) {
   const status = {msg: 'Saving artifacts', id: 'lh:assetSaver:saveArtifacts'};
   log.time(status);
   fs.mkdirSync(basePath, {recursive: true});
-  rimraf.sync(`${basePath}/*${traceSuffix}`);
-  rimraf.sync(`${basePath}/${artifactsFilename}`);
+
+  // Delete any previous artifacts in this directory.
+  const filenames = fs.readdirSync(basePath);
+  for (const filename of filenames) {
+    if (filename.endsWith(traceSuffix) || filename.endsWith(devtoolsLogSuffix) ||
```

Comment:
It's not clear why we weren't deleting `latest-run/*.devtoolslog.json` before. Maybe we just wrote this before devtoolslogs as files existing and didn't update?

It leads to weird situations since `loadArtifacts()` will happily load any file in the directory ending with `devtoolslog.json`, so you can easily get devtoolsLogs from previous runs (like `pageLoadError-defaultPass.devtoolslog.json`) in the artifacts of a brand new `-A` run. I don't think we currently have any code that would automatically do anything with logs with non-default names, but still weird for them to be mixed in there